### PR TITLE
Merge magnetic forward functions for dipoles

### LINKED ISSUE #450: Update magnetic forward functions for dipoles
**Description of the desired feature:**

After solving #446 and #447 with #448, we should apply similar changes to the dipole magnetic forward functions.

In a nutshell:
- merge the `dipole_magnetic` and `dipole_magnetic_component` functions into a single one,
- change the shape and type of the expected `magnetic_moment` argument: it should be a tuple of three arrays: `mag_moment_e`, `mag_moment_n`, `mag_moment_u`.

**Are you willing to help implement and maintain this feature?** 

Sure!

### PR DESCRIPTION
**Pull request by santisoler**
Merge the two forward functions for magnetic fields of dipoles into a single one named `dipole_magnetic` that includes a new `field` argument. Make the `dipole_magnetic` function to ask magnetic moment vectors as a tuple of three arrays, where each one of these arrays. Update tests accordingly. Update the user guide page about forward modelling of prisms.

**Relevant issues/PRs:**

Fixes #450

Implements the same interface as #448

### CODE REVIEW

**Comment by santisoler** [comment]
@indiauppal I think this is ready to be merged. Would you like to take a look at it and review it? Feel free to say no if you don't feel like it, no pressure!

**Review by indiauppal — COMMENTED** [review]
Looks good, ready to merge @santisoler

File: harmonica/_forward/dipole.py [hunk 1]
```diff
@@ -54,137 +86,207 @@ def dipole_magnetic(
...
+    progressbar : bool
+        If True, a progress bar of the computation will be printed to standard
+        error (stderr). Requires :mod:`numba_progress` to be installed.
 
     Returns
     -------
-    magnetic_field :  array
-        Array containing the magnetic field component in :math:`nT`.
+    magnetic_component : arrays
+        Array containing the desired magnetic component.
```
Comment (santisoler) [inline harmonica/_forward/dipole.py:248]:
```suggestion
        Array containing the desired magnetic component in :math:`nT`.
```

**Comment by santisoler** [comment]
Thanks a lot @indiauppal! Just pushed two minor changes I caught, nothing critical. After CI passes I'll merge this. Hope these changes turn out to be helpful.

**Pull request merged**